core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -84,11 +82,13 @@ public void addDayActivity(DayActivity dayActivity)
 			throw new IllegalStateException("Week is already full (" + dayActivities.size() + " days present)");
 		}
 
+		dayActivity.setWeekActivity(this);
 		dayActivities.add(dayActivity);
 	}
 
 	public void removeAllDayActivities()
 	{
+		dayActivities.forEach(da -> da.setWeekActivity(null));
```

Comment:
Well, the delete is implicit here, so you can leave it with this but then you should maybe add a comment that this works because of orphan removal. It doesn't matter much.